src/kernel/mocaml.ml

```diff
@@ -35,8 +35,12 @@ let setup_reader_config config =
   let open Mconfig in
   let open Clflags in
   let ocaml = config.ocaml in
-  let to_compilation_unit name = Some (Compilation_unit.of_string name) in
-  Env.set_unit_name (Mconfig.unitname config |> to_compilation_unit);
+  let to_unit_info name =
+    Some (name |>
+          Compilation_unit.of_string |>
+          Unit_info.make_dummy ~input_name:name)
```

Comment:
I think this won't break anything, but using `make_dummy` scares me a bit. It seems like the type of thing that may cause something to break down the line. If we want to call `Unit_info.make_with_known_compilation_unit`, we can use `config.query.filename` to fill `source_file`.

I'm not sure about `intf_or_impl` - I don't know if Merlin knows that at this point (I'm not sure if Merlin decides this based on solely the filename or whether it's able to parse successfully). And I don't know that the `file_prefix` argument is for.

Maybe @lukemaurer can chime in on the significance of using `make_dummy` here, and what the deal is with `file_prefix`?

Reply:
I seem to have lost my permissions to push to this repo. @lukemaurer can you look at the below commit: https://github.com/liam923/merlin/commit/4123c5a9e3e07c735e6aab92e8d282f2cb891628

Specifically, I'm not sure what to pass for the `file_prefix` to `Unit_info.make_with_known_compilation_unit`.

Reply:
Aha I've gotten my permissions back. Commit 4123c5a is pushed here now.